### PR TITLE
[pcapplusplus] Update to 22.11

### PR DESCRIPTION
**Pull request by seladb**
Update PcapPlusPlus port to v22.11

- #### What does your PR fix?
  Update PcapPlusPlus to v22.11

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  <all / linux, windows, ...>, <Yes/No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by Adela0814** [comment]
Please get [failure logs here](https://dev.azure.com/vcpkg/public/_build/results?buildId=80775&view=artifacts&pathAsName=false&type=publishedArtifacts). Please ping me if you need any help.

**Comment by seladb** [comment]
> Please get [failure logs here](https://dev.azure.com/vcpkg/public/_build/results?buildId=80775&view=artifacts&pathAsName=false&type=publishedArtifacts). Please ping me if you need any help.

Thank you @Adela0814 ! I can't figure out what is the error... I don't seem to have permission to download the log files you share, and from the CI log I can't understand why it fails: https://dev.azure.com/vcpkg/public/_build/results?buildId=80775&view=logs&jobId=7b75bd19-17d3-53d4-00fd-23f1a49a8ba4&j=7b75bd19-17d3-53d4-00fd-23f1a49a8ba4&t=a23dba4e-3a62-59dc-a73c-a7222676d7a4

**Comment by Adela0814** [comment]
Failure log:
[install-x64-osx-dbg-out (2).log](https://github.com/microsoft/vcpkg/files/9999009/install-x64-osx-dbg-out.2.log)

**Comment by Adela0814** [comment]
Convert this PR to draft since there is no progress. Please ping us if this PR is ready for review again.

**Comment by seladb** [comment]
@Adela0814 I'm sorry for the delayed response. Looking at the failure logs I'm not sure what's the issue. I see errors like this:
```text
/Users/vagrant/Data/buildtrees/pcapplusplus/src/v22.11-e3390d66ff.clean/Packet++/header/FtpLayer.h:62:8: error: reference to enumeration must use 'enum' not 'enum class' [-Welaborated-enum-class]
                enum class FtpCommand : int
                     ^~~~~~

...

/Users/vagrant/Data/buildtrees/pcapplusplus/src/v22.11-e3390d66ff.clean/Packet++/header/FtpLayer.h:288:8: error: reference to enumeration must use 'enum' not 'enum class' [-Welaborated-enum-class]
                enum class FtpStatusCode : int
                     ^~~~~~
```

And then a lot of warnings regarding c++11 extensions, for example:

```text
/Users/vagrant/Data/buildtrees/pcapplusplus/src/v22.11-e3390d66ff.clean/Packet++/header/FtpLayer.h:62:8: warning: scoped enumerations are a C++11 extension [-Wc++11-extensions]
                enum class FtpCommand : int
                     ^
...

/Users/vagrant/Data/buildtrees/pcapplusplus/src/v22.11-e3390d66ff.clean/Packet++/header/FtpLayer.h:288:8: warning: scoped enumerations are a C++11 extension [-Wc++11-extensions]
                enum class FtpStatusCode : int
                     ^
```

This fails in `install-x64-osx-dbg-out.log`.
Maybe there is a missing declaration somewhere that this library requires C++11?

FYI, this library builds successfully on many platforms in the project's CI, including macOS x64: https://github.com/seladb/PcapPlusPlus/actions/runs/3627962304

**Comment by dg0yt** [comment]
vcpkg uses a vendored `CMakeLists.txt`. So updating the port requires updating  `CMakeLists.txt`. Adding the C++11 requirment maybe just one change which is needed. 
Until eventually using upstream's native build system.

**Comment by dg0yt** [comment]
@seladb I see you are reviewing https://github.com/seladb/PcapPlusPlus/pull/944. :+1:

**Comment by seladb** [comment]
> vcpkg uses a vendored `CMakeLists.txt`. So updating the port requires updating `CMakeLists.txt`. Adding the C++11 requirment maybe just one change which is needed. Until eventually using upstream's native build system.

Thank you for your reply @dg0yt ! The current PcapPlusPlus version (22.11) uses the same build system that was used in previous PcapPlusPlus versions (which is a custom build system, not CMake) and the current port works well. The biggest change in 22.11 is moving to C++11, nothing in PcapPlusPlus build system changed.

The PR you are referring to https://github.com/seladb/PcapPlusPlus/pull/944 is not yet merged, and anyway, it will only get into the next version of PcapPlusPlus (which might be released in a few months from now).

I'm trying to update the port for this version. Any idea why the build is failing?

**Comment by seladb** [comment]
@egecetin it seems your fix doesn't work 😢

**Comment by egecetin** [comment]
Looks like this time linux/windows also fail right?

File: ports/pcapplusplus/portfile.cmake [hunk 1]
```diff
@@ -13,6 +13,7 @@ file(COPY "${CURRENT_PORT_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
+    -DCMAKE_CXX_STANDARD 11
```
Comment (dg0yt) [inline ports/pcapplusplus/portfile.cmake:16]: `-DCMAKE_CXX_STANDARD=11`

But again: this belongs into `ports/pcapplusplus/CMakeLists.txt`.

Comment (seladb) [reply ports/pcapplusplus/portfile.cmake:16]: @dg0yt there is no such file in pcapplusplus port 🤔

Comment (dg0yt) [reply ports/pcapplusplus/portfile.cmake:16]: `ports/pcapplusplus/CMakeLists.txt` in vcpkg, not upstream. It copied to the source dir just before `vcpkg_cmake_configure`.

Comment (seladb) [reply ports/pcapplusplus/portfile.cmake:16]: So what is the change needed in this PR to enable C++11 and make the tests pass?
I'm sorry for my poor vcpkg knowledge

**Comment by seladb** [comment]
@Adela0814 @dg0yt the PR is ready for review. Can you please take a look?

**Comment by Adela0814** [comment]
@seladb Please fix the conflicts.

**Comment by seladb** [comment]
> @seladb Please fix the conflicts.

@Adela0814 done!

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**